### PR TITLE
fix: incorrect currency used for market data in various parts of the app

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

* Fixes 3 cases where we're using USD market data in place of user-currency market data
* Improves naming of market data related selectors to prevent confusion in the future

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

NA

Marking as high risk as an error in renaming here could have big consequences around the app. The changes in this PR are mostly renaming though, and the only logical changes are the 3 fixes marked.

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

Check market data and prices around the app are looking correct in non-usd currencies, specifically:
* Trade confirm summary of asset amounts in non-USD fiat
* Staking positions  in non-USD fiat
* LP positions  in non-USD fiat

### CODE REVIEW

**Comment by woodenfurniture** [comment]
* **#6361** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6361?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @woodenfurniture and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/components/EarnDashboard/components/PositionDetails/StakingPositionsByProvider.tsx [hunk 1]
```diff
@@ -55,13 +55,13 @@ const calculateRewardFiatAmount: CalculateRewardFiatAmount = ({
...
   assets,
-  marketData,
+  marketDataUserCurrency,
 }) => {
   if (!rewardAssetIds) return 0
   return Array.from(rewardAssetIds).reduce((sum, assetId, index) => {
     const asset = assets[assetId]
     if (!asset) return sum
-    const marketDataPrice = bnOrZero(marketData[assetId]?.price)
+    const marketDataPrice = bnOrZero(marketDataUserCurrency[assetId]?.price)
```
Comment (woodenfurniture) [inline src/components/EarnDashboard/components/PositionDetails/StakingPositionsByProvider.tsx:64]: first fix, this was pulling in USD data which it should be user currency

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/AssetSummaryStep.tsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ export const AssetSummaryStep = ({
   const {
     number: { toCrypto, toFiat },
   } = useLocaleFormatter()
-  const fiatPriceByAssetId = useAppSelector(selectCryptoMarketData)
+  const marketDataUserCurrency = useAppSelector(selectCryptoMarketDataUserCurrency)
```
Comment (woodenfurniture) [inline src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/AssetSummaryStep.tsx:29]: second fix, this was pulling in USD data which it should be user currency

File: src/components/EarnDashboard/components/PositionDetails/LpPositionsByProvider.tsx [hunk 3]
```diff
@@ -50,7 +50,7 @@ export const LpPositionsByProvider: React.FC<LpPositionsByProviderProps> = ({ id
     dispatch,
   } = useWallet()
   const assets = useAppSelector(selectAssets)
-  const marketData = useAppSelector(selectCryptoMarketDataSortedByMarketCapUsd)
+  const marketDataUserCurrency = useAppSelector(selectCryptoMarketDataSortedByMarketCapUserCurrency)
```
Comment (woodenfurniture) [inline src/components/EarnDashboard/components/PositionDetails/LpPositionsByProvider.tsx]: Another fix, we want user currency here

File: src/components/EarnDashboard/components/PositionDetails/StakingPositionsByProvider.tsx [hunk 4]
```diff
@@ -28,7 +28,7 @@ import { getMetadataForProvider } from 'state/slices/opportunitiesSlice/utils/ge
 import {
   selectAggregatedEarnUserStakingOpportunitiesIncludeEmpty,
   selectAssets,
-  selectCryptoMarketData,
+  selectCryptoMarketDataUserCurrency,
```
Comment (gomesalexandre) [inline src/components/EarnDashboard/components/PositionDetails/StakingPositionsByProvider.tsx:31]: 👀

**Review by gomesalexandre — COMMENTED** [review]
First pass, renaming looks sane minus `selectCryptoMarketDataUserCurrency` which is quite confusing as discussed, though non-blocking as it's already confusing enough as is.
USD references are clearly denoted as such naming-wise now, and same goes for user currency ones

**Review by gomesalexandre — APPROVED** [review]
Tested locally (left) against develop (right)

- Crypto receive amount is now prorated with the *user currency*, not the USD rate, i.e is similar to the fiat input amount

image
image

- Staking positions are prorated with *user currency* USD rate

image

- LP positions are prorated with *user currency* USD rate
 
image

**Review by 0xApotheosis — APPROVED** [review]

**Pull request merged**